### DISCUSSION TITLE
update collection remove problem

### DISCUSSION THREAD

**Discussion by lineCode** [mikro-orm/nestjs#116] [General]
**Describe the bug**
// resets the addresses collection to a single item
wrap(user).assign({ addresses: [new Address(...)] });

this will insert new items before, and delete old items after, if the has unique index, the may throw conflict error

**Expected behavior**
delete old items first, and insert new items after

**Versions**

| Dependency | Version |
| - | - |
| node | v16.14.0 |
| typescript | 4.3.5 |
| mikro-orm | 5.6.14 |
| your-driver | mysql |

**Comment by B4nan** [comment]
I can't help without a complete reproduction, there are many places in the test suite that are already doing what you say, and it works just fine.

The problem of [deleting items early is addressed too](https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/unit-of-work/UnitOfWork.ts#L492-L508), maybe the detection does not work, or maybe it is some edge case with working with collections. But again, I need to see the failing code myself.

I will move this to discussions before we find a failing reproduction, in the end the problem is most probably not nestjs specific and we should have the issue elsewhere.
